online-store/src/components/DescriptionPage.ts

```diff
@@ -0,0 +1,175 @@
+        </div>
+      </div>
+    </div>
+  </main>
+  <footer class="footer">
+    <div class="github-links">
+      <img
+        class="footer__svg"
+        src="https://raw.githubusercontent.com/YuliyaBondar/christmas-data/main/assets/svg/github.svg"
+        alt="rs_school_js"
+      />
+      <a href="https://github.com/YuliyaBondar">YuliyaBondar</a>
+      <a href="https://github.com/poznerrr">poznerrr</a>
+    </div>
+    <p>2022</p>
+    <a href="https://rs.school/js/"
+      ><img
+        class="footer__svg"
+        src="https://raw.githubusercontent.com/YuliyaBondar/christmas-data/main/assets/svg/rss.svg"
+        alt="rs_school_js"
+    /></a>
+  </footer>
+        `;
+    return '1';
+  }
+
+  async renderCrumbs() {
+    const crumbs = document.querySelector('.crumbs');
+    const globalCrumb = document.createElement('div');
+    globalCrumb.classList.add('crumb');
+    globalCrumb.innerText = `LEGO`;
+    crumbs?.appendChild(globalCrumb);
+    const firstCrumb = document.createElement('div');
+    firstCrumb.classList.add('crumb');
+    firstCrumb.innerText = ` > `;
+    crumbs?.appendChild(firstCrumb);
+    const themeCrumb = document.createElement('div');
+    themeCrumb.classList.add('crumb');
+    themeCrumb.innerText = `${this.product.category}`;
+    crumbs?.appendChild(themeCrumb);
+    const secondCrumb = document.createElement('div');
+    secondCrumb.classList.add('crumb');
+    secondCrumb.innerText = ` > `;
+    crumbs?.appendChild(secondCrumb);
+    const nameCrumb = document.createElement('div');
+    nameCrumb.classList.add('crumb');
+    nameCrumb.innerText = `${this.product.name}`;
+    crumbs?.appendChild(nameCrumb);
+  }
+
+  async addEvents() {
+    await this.renderCrumbs();
+
+    const images = document.querySelectorAll('.mainProduct__img');
+    images.forEach((img) => {
+      img.addEventListener('mouseover', () => this.addresize(img));
+      img.addEventListener('mouseleave', () => this.removeresize(img));
+    });
+
+    this.buttonChanger();
+    const btnAdd = document.querySelector('.mainProduct__add');
+    btnAdd?.addEventListener('click', () => {
+      this.goToBasket();
+    });
+
+    const btnRemove = document.querySelector('.mainProduct__remove');
+    btnRemove?.addEventListener('click', () => {
+      this.removeFromBasket();
+    });
+
+    const btnBuy = document.querySelector('.mainProduct__buy');
+    btnBuy?.addEventListener('click', () => {
+      this.buyNow();
+    });
+  }
+
+  addresize(element: Element) {
+    element.classList.add('resize-img');
+  }
+  removeresize(element: Element) {
+    element.classList.remove('resize-img');
+  }
+
+  buttonChanger() {
+    const btnAdd = document.querySelector('.mainProduct__add');
+    const btnRemove = document.querySelector('.mainProduct__remove');
+    if (AppState.instance.state.basket.orders.filter((item) => item.legoItem === this.product.id).length === 0) {

```

Comment:
ну или как луше обозвать переменную - вы больше в контексте. но такие условия оставлять нельзя)

```suggestion
const isNoOrders = AppState.instance.state.basket.orders.filter((item) => item.legoItem === this.product.id).length === 0;
    if (isNoOrders) {

```